edb/pgsql/compiler/relgen.py

```diff
@@ -3483,6 +3483,45 @@ def process_set_as_exists_expr(
     return new_stmt_set_rvar(ir_set, stmt, ctx=ctx)
 
 
+@_special_case('std::json_object_pack')
+def process_set_as_json_object_pack(
+        ir_set: irast.Set, stmt: pgast.SelectStmt, *,
+        ctx: context.CompilerContextLevel) -> SetRVars:
+    expr = ir_set.expr
+    assert isinstance(expr, irast.FunctionCall)
+
+    with ctx.subrel() as subctx:
```

Comment:
I think that there are a bunch of places where we create subrels without really needing to. We should try to clean some of that up.

There are also definitely a lot of places where we create subrels for purely technical internal reasons, because something or another in the compiler (often the path map) is connected to subrels. (https://github.com/edgedb/edgedb/blob/e1f0f06be29a35600dba0cb906d6a53d051ad8ff/edb/pgsql/compiler/relgen.py#L1877-L1887 for an example)

Reply:
> `stmt` is always just `ctx.rel`

I would suggest that we don't also pass `stmt` to all the functions then. It would cleanup the signatures quite a lot and only make us use two chars more :D